odoo/tools/float_utils.py

```diff
@@ -191,8 +191,7 @@ def float_split_str(value, precision_digits):
     """
     value = float_round(value, precision_digits=precision_digits)
     value_repr = float_repr(value, precision_digits)
-    units, cents = value_repr.split('.')
-    return units, cents
+    return tuple(value_repr.split('.')) if precision_digits else (value_repr, '')
```

Comment:
Hmm, `float_split()` would still crash with 0 digits:

```py
>>> float_split(3.0, 0)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/odoo/odoo/tools/float_utils.py", line 204, in float_split
    return int(units), int(cents)
ValueError: invalid literal for int() with base 10: ''
```

Though if we fix it, can we obtain consistent behavior between `float_split_str()` and `float_split()`, considering that the latter must always return 2 integers?

```py
float_split(3.14, 0) == (3, 0) # must return 2 ints
float_split_str(3.14, 0) == ('3', '') # weird, huh?

```

Either way, this deserves a mention in the docstrings, as the current explanation doesn't cut it.
And `float_split(f, 0)` should be covered with a proper test too, as we can see.

Reply:
@odony Indeed. It was actually already crashing before, just not at the same place (float_split_str instead of float_split). I'll add something in the docsting and return None as the second value instead of crashing. I guess it's the cleanest way.

Reply:
> I'll add something in the docsting and return None as the second value instead of crashing. I guess it's the cleanest way.

The problem with returning `None` is that it changes the documented signature `:rtype: tuple(int, int)`. 

We should aim for the least-surprising result for the caller, taking into account the kind of code where that result is expected be used. For `float_split()`, that would be some arithmetic calculations, where the caller is unlikely to except the appearance of `None` as the decimal digit.

So `0` seems to be a less surprising result in this case, even if that is inconsistent with what `float_split_str` does. On second thought I guess that's ok, there are other differences caused by the different return types and expectations. For example the str version will return `'0'` or `'00'` as decimal part depending on the number of decimal digits, whereas the int version gives the same result.

Reply:
I was editing my message just as you were posting this one ^^ 0 is fine to me, indeed it makes more sense for arithmetic reasons.